tsconfig.json

```diff
@@ -9,6 +9,9 @@
     "noEmit": true,
     "noErrorTruncation": true,
     "noUncheckedIndexedAccess": true,
+    "paths": {
+      "@metamask/utils/node": ["./node_modules/@metamask/utils/dist/types/node"]
```

Comment:
This is here because we've instructed TypeScript with to use its "classic" module resolution algorithm via `moduleResolution` = `node`. Unfortunately, this means that it's not able to import `@metamask/utils/node`, which is a special import in the `@metamask/utils` that grants access to Node.js-specific functions. The `@metamask/utils/node` path works because this package defines an `exports` field in its manifest: https://github.com/MetaMask/utils/blob/fb9440203099538670a4af95da16265f878f2cc9/package.json#L30. This setting works around that by mapping `@metamask/utils/node` to `dist/types/node.d.ts` inside of the `@metamask/utils`.

Unfortunately, doing this prohibits this package from ever being used as a library. At the moment, this is okay, because we're only using it as a CLI, which means it runs in Node, and Node knows how to read `exports` fields. But if we ever do want to expose an API, we'll have to come up with a better solution.

Reply:
>and we're doing that because it works best with the way that the extension compiles JavaScript

Could you elaborate on this part? I'm curious to know what the incompatibility is here. Using modern module resolution would simplify things

Reply:
Sorry, that doesn't make sense. I was really trying to say "it works best with how we compile all libraries".

First, if we change `moduleResolution` to `node` (and just this, nothing else), this will not work once we start using TypeScript 5.2, which [revised the rules around `module` and `moduleResolution`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html#module-and-moduleresolution-must-match-under-recent-nodejs-settings) such that `moduleResolution` = `nodenext` must be paired with `module` = `nodenext`.

I was playing around with [another branch that compiles all of my changes for this project into one](https://github.com/MetaMask/module-lint/tree/add-initial-code) and I swear that running `yarn build` caused some kind of build error with another library. I can't seem to replicate it now.

So, I suppose I could fix this at the `tsconfig.json` level.

Reply:
Hmm OK, I'd like to dig into this further but it doesn't need to block this PR